subsys/bluetooth/host/gatt.c

```diff
@@ -4122,11 +4143,20 @@ static void gatt_prepare_write_rsp(struct bt_conn *conn, uint8_t err,
 	}
 
 	len = length - sizeof(*rsp);
+
+	data_valid = memcmp(params->data, rsp->value, len) != 0;
```

Comment:
Shouldn't use len before we have checked it against the params->length below. This should be moved down.

Reply:
Of course, fixed